src/rust/bitbox02-rust/src/hww/api/bitcoin/policies.rs

```diff
@@ -207,6 +244,34 @@ impl<'a> ParsedPolicy<'a> {
 
         Ok(())
     }
+
+    /// Derive the witness script of the policy derived at a receive or change path.
+    /// If is_change is false, the witness script for the receive address is derived.
+    /// If is_change is true, the witness script for the change address is derived.
+    /// Example: wsh(and_v(v:pk(@0/**),pk(@1/<20;21>/*))) derived using `is_change=false, address_index=5` derives
+    /// wsh(and_v(v:pk(@0/0/5),pk(@1/20/5))).
+    /// The same derived using `is_change=true` derives: wsh(and_v(v:pk(@0/1/5),pk(@1/21/5)))
+    #[allow(dead_code)] // will be used soon in the creation of receive addresses
+    pub fn witness_script(&self, is_change: bool, address_index: u32) -> Result<Vec<u8>, Error> {
+        match self {
+            Self::Wsh(Wsh {
+                policy,
+                miniscript_expr,
+            }) => {
+                let mut translator = WalletPolicyPkTranslator {
+                    keys: policy.keys.as_ref(),
+                    is_change,
+                    address_index,
+                };
+                let miniscript_expr = match miniscript_expr.translate_pk(&mut translator) {
```

Comment:
super minor nit: if I understood it correctly, `miniscript_expr` is a bit misleading, since it has been translated into a witness bitcoin script. Maybe change to `witness_script`?

Reply:
It has not been translated to a witness script yet. The `.encode()` function does that. `translate_pk` transforms from `Miniscript<String, ...>` to `Miniscript<bitcoin::PublicKey, ...>`, so it is the same AST, just with the replaced pubkeys.